client/src/WebSocket.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react'
+
+export interface WebSocketProviderProps {
+  url: string
+  children: React.ReactNode
+}
+
+export type WebSocketSend = (message: string) => void
+
+const WebSocketContext = React.createContext<WebSocket | null>(null)
+
+export const useWebSocket = (): WebSocket | null => {
+  return React.useContext<WebSocket | null>(WebSocketContext)
+}
+
+export const useSend = (): WebSocketSend => {
+  const webSocket = useWebSocket()
+  return React.useCallback<WebSocketSend>(
+    (message: string) => {
+      if (webSocket) webSocket.send(message)
+    },
+    [webSocket]
+  )
+}
+
+export type WebSocketStateConsumer = (message: object) => void
+
+export const useWebSocketState = (dispatcher: WebSocketStateConsumer) => {
```

Comment:
Change this to standard props like `onMessageReceived`. The message can then be handled by something higher up. The only thing that should be a hook now is `useSend`